plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -630,7 +639,34 @@ public void replaceTable(String databaseName, String tableName, Table newTable,
     @Override
     public void renameTable(String databaseName, String tableName, String newDatabaseName, String newTableName)
     {
-        throw new TrinoException(NOT_SUPPORTED, "Table rename is not yet supported by Glue service");
+        boolean newTableCreated = false;
+        try {
+            Optional<Table> table = getTable(databaseName, tableName);
+            if (table.isEmpty()) {
+                throw new TableNotFoundException(schemaTableName(databaseName, tableName));
+            }
+            TableInput tableInput = GlueInputConverter.convertTable(table.get())
+                    .withName(newTableName);
+            CreateTableRequest createTableRequest = new CreateTableRequest()
+                    .withDatabaseName(newDatabaseName)
+                    .withTableInput(tableInput);
```

Comment:
Q Does this request preserve all information about the table definition? e.g. catalog id, partition indexes. 

Even if the current answer is yes, new information will lost if Glue add new fields to `CreateTableRequest` in the future. It would be better to leave a code comment in my opinion.

Reply:
What kind of comment? Could you please propose something?
 I assume that `GlueInputConverter.convertTable` knows what it is doing and we have to keep it up to date anyway. 
Also we have similar case for iceberg.

Reply:
> What kind of comment? Could you please propose something?

It's just a question rather than suggestions. I know the question is for existing method. Sorry about that. 

I think catalog id is probably fine. I still have small concern about edge cases though. For instance, I'm not sure what happens if the resource comes from cross-account access https://docs.aws.amazon.com/glue/latest/dg/cross-account-access.html.

I confirmed using `GlueInputConverter.convertTable` will lost some informations. e.g. retention. 
We can reproduce by setting a table retention via Glue CLI `update-table` and rename in Trino. The value becomes 0 after the rename.

Reply:
hmm ok I see, I will take a look what can be done

Reply:
Don't we need to copy partition indexes as well? It disappeared as far as I tested.